pkgs/applications/networking/browsers/firefox/common.nix

```diff
@@ -18,7 +18,7 @@
 , hunspell, libevent, libstartup_notification
 , libvpx
 , libwebp
-, icu, libpng, glib, pciutils
+, icu, libpng_apng, glib, pciutils
```

Comment:
This isn't actually needed, since `libpng_apng = libpng.override { apngSupport = true; };` and `apngSupport ? true` in `pkgs/development/libraries/libpng/default.nix`.

Also:
```
  firefoxPackages = recurseIntoAttrs (callPackage ../applications/networking/browsers/firefox/packages.nix {
    callPackage = newScope {
      inherit (rustPackages) cargo rustc;
      libpng = libpng_apng;
      gnused = gnused_422;
      inherit (darwin.apple_sdk.frameworks) CoreMedia ExceptionHandling
                                            Kerberos AVFoundation MediaToolbox
                                            CoreLocation Foundation AddressBook;
      inherit (darwin) libobjc;
    };
  });
```
same is true for `thunderbirdPackages`.

8164d28315eae1b0cb4aeb26f7dc3334e6ff9e69 is the commit that enabled apng by default in 2015. So how about you drop this commit from this PR and I drop `libpng_apng` from nixpkgs?

Reply:
The same is true for `libjpeg_turbo` then.

Reply:
Reverted to `libpng` and `libjpeg`. Cleaned out the top-level mess. Stopped passing the full rustPackages in favor of that inherit statement.